### DISCUSSION TITLE
Subscribe  with RemotConnection not fired

### DISCUSSION THREAD

**Discussion by TuniDev-HUB** [DevrexLabs/memstate#130] [General]
Greeting,
Thank you for your great JOB.

The system did not fired this event. I guess this is empty(RemoteClient) :
        private readonly Dictionary<Type, Action<Event>> _eventHandlers;

////////////////////////////////////////////
            Config.Reset();
            Config.Current.UseInMemoryFileSystem();

            var cfg = Config.Current;
            var clientSettings = cfg.GetSettings<ClientSettings>();
            clientSettings.Host = IPAddress.Loopback.ToString();
            clientSettings.Port = 5505;
            clientSettings.Type = ConnectionType.Remote;

            _client = Client.For<InMemoryCache>().GetAwaiter().GetResult();
            var eventsReceived = new List<Event>();
            await _client.Subscribe<Added>(e => doHandlerEvent(e));

NB: I used this instrection RaiseEvent(new Added(_data));

**Comment by rofr** [comment]
Yes, this looks like a bug, thanks for researching and reporting. I recall seeing the bug during a workshop but there hasn't been any interest from users in getting it fixed. Are you evaluating memstate or using it in production? Please tell about your use case!

**Reply by TuniDev-HUB** [reply]
I'm in evaluation step. We have monolithic application that use In-memory Cache inside it. 
I attempt to create In-Memory cache server (to migrate our App to distributed system) at the first step. That's why I experiment your subscript system that dispatch event from server to All clients.

**Reply by rofr** [reply]
Interesting! Replicated cache is a very common use case. There is a much easier way to achieve this than running the cache in it's own remote process with clients in each application. Instead you should run a local engine inside each application process all using the same backing store.
